lib/Web/Server.hs

```diff
@@ -28,19 +27,149 @@ import Web.API
 import Backend.Interpreter (runBackend)
 import Coffer.Directory (Directory)
 import Coffer.Path (EntryPath, Path, QualifiedPath)
+import Coffer.PrettyPrint
+  (PrettyPrintMode(WebAPI), buildCopyOrRenameResult, buildCreateError, buildDeleteFieldResult,
+  buildDeleteResult, buildSetFieldResult, buildTagResult, buildViewResult)
+import Data.Aeson
+import Data.Aeson.Casing
+import Data.Aeson.TH
 import Data.Functor ((<&>))
 import Data.Set qualified as Set
-import Fmt (pretty)
+import Fmt (Builder, pretty, unlinesF)
+import GHC.Generics (Generic)
 import Servant.API
-import Servant.Client
 import Servant.Server
 import Web.Types (NewEntry(NewEntry), NewField(NewField))
 
-url :: BaseUrl
-url = BaseUrl Http "127.0.0.1" 8200 ""
+data CofferServerError = CofferServerError
+  { cseError :: Text
+  , cseCode :: Int
+  }
+  deriving stock (Show, Generic)
+deriveJSON (aesonPrefix camelCase) ''CofferServerError
+
+cofferServerErrorToBytes :: Text -> Int -> Bytes.Lazy.ByteString
+cofferServerErrorToBytes msg code = encode [CofferServerError msg code]
+
+resultToText :: (PrettyPrintMode -> a -> Builder) -> a -> Text
+resultToText f res = pretty $ f WebAPI res
+
+addApplicationJSONHeader :: ServerError -> ServerError
+addApplicationJSONHeader err@(ServerError _ _ _ headers) =
+  err { errHeaders = applicationJSONHeader : headers }
+  where
+    applicationJSONHeader = ("Content-Type", "application/json;charset=utf-8")
+
+throwCofferServerError :: ServerError -> Int -> Text -> Handler a
+throwCofferServerError err code msg =
+  throwError $ addApplicationJSONHeader err { errBody = cofferServerErrorToBytes msg code }
+
+throwCofferServerErrors :: ServerError -> [(Int, Text)] -> Handler a
+throwCofferServerErrors err codesAndMsgs = do
+  let cofferServerErrors = uncurry (flip CofferServerError) <$> codesAndMsgs
+  throwError $ addApplicationJSONHeader err { errBody = encode cofferServerErrors }
```

Comment:
`addApplicationJSONHeader` is only used in one place, so i think it would make sense to simplify things and inline it here:

```suggestion
  throwError err
    { errBody = encode cofferServerErrors
    , errHeaders = ("Content-Type", "application/json;charset=utf-8") : errHeaders err
    }
```

Reply:
Done